e2etest/clean/clean_distributor_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package clean
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"sort"
+	"strconv"
+	"testing"
+)
+
+func TestCleanDistributor(t *testing.T) {
+	//set up aws go sdk config
+	awsConfig := aws.Config{
+		Region: aws.String("us-west-2"),
+	}
+	testSession, err := session.NewSession(&awsConfig)
+	if err != nil {
+		t.Fatalf("Error creating session %s", err.Error())
+	}
+	// set up aws go sdk ssm client
+	ssmClient := ssm.New(testSession, &awsConfig)
+
+	//get list of documents with versions
+	documentName := "testAWSDistroOTel-Collector"
+	listDocumentVersionInput := ssm.ListDocumentVersionsInput{Name: &documentName}
+	listDocumentVersionsOutput, err := ssmClient.ListDocumentVersions(&listDocumentVersionInput)
+	if err != nil {
+		t.Fatalf("Error getting document list %s", err.Error())
+	}
+	t.Logf("Document Versions %s", listDocumentVersionsOutput.String())
+	//get document versions
+	var versions []int
+	for i := range listDocumentVersionsOutput.DocumentVersions {
+		//can't delete the default version
+		if *listDocumentVersionsOutput.DocumentVersions[i].IsDefaultVersion {
+			continue
+		}
+		version, err := strconv.Atoi(*listDocumentVersionsOutput.DocumentVersions[i].DocumentVersion)
+		if err != nil {
+			t.Fatalf("Error getting document list %s", err.Error())
+		}
+		versions = append(versions, version)
+	}
+	//sort document versions from oldest to newest
+	sort.Ints(versions[:])
+	t.Logf("Versions %v", versions)
+	//delete all documents except the 5 latest
+	for i := 0; i < len(versions)-5; i++ {
+		version := strconv.Itoa(versions[i])
```

Comment:
```suggestion
	for _, v := range versions[:len(versions)-5] {
		version := strconv.Itoa(v)
```

Here slicing does make sense and also avoids any confusion about what the index variable is for.